test/functional/wallet_encryption.py

```diff
@@ -46,6 +48,25 @@ def run_test(self):
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrase, '', 1)
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrasechange, '', 'ff')
 
+        # Make sure there are no unencrypted key records
+        # Use the wallet tool's dump mechanism to get all of the records to inspect them.
+        self.nodes[0].unloadwallet(self.default_wallet_name)
+        dump_file = os.path.join(self.nodes[0].datadir, "wallet.dump")
```

Comment:
Seems to not be a problem in the CI but.. for some reason, I'm unable to run this locally. Python 3.11

```python
dump_file = os.path.join(self.nodes[0].datadir, "wallet.dump")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "<frozen posixpath>", line 76, in join
TypeError: expected str, bytes or os.PathLike object, not AuthServiceProxyWrapper
```

Reply:
ok, found the problem. This should use `datadir_path` not `datadir`:

``` python3
self.nodes[0].datadir_path / "wallet.dump"
```

Reply:
Fixed